luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -34,7 +34,7 @@
      * @param resourceDictionaries  The dictionaries to initialize the industrial park with.
      * @param dimensionFactories The map of factories for creating dimensions from external config
      */
-    protected LuthierIndustrialPark(
+    public LuthierIndustrialPark(
```

Comment:
Why make this `public`? The intent is that people use LuthierIndustrialPark.Builder to
build an instance, rather than building one with the constructor. For one thing, the Builder makes it
a lot easier to add custom factories without having to also pull in all the local stuff. If you want
to add your own dimensionFactory, it's as easy as:
```java
LuthierIndustrialPark park = new LuthierIndustrialPark.Builder()
.withDimensionFactory("testDimensionFactory", new TestDimensionFactory<>())
.build();
```

No need for making the constructor public.

Reply:
The problem here is that the 
`LuthierBinderFactory::getConfigurationLoader()` was using a ctor directly. I have changed the implementation there to use a Builder as well. 
To make it really work, I think we should make the Builder a **static** nested class of LuthierIndustrialPark, since we want it to be called before an instance of it's parent is initialized. 

The current ctor for Builder requires a preexisting ResourceDictionaries instance. 
To make a `Builder()` ctor is surprisingly non-trivial, because we cannot just set its resourceDictionaries as a `new ResourceDictionaries`, as that can be un-initialized. (The initializer is protected in the LuthierBinderFactory)

Since this is more intertwined with the design behind `LuthierREsourceDictionaries`, I think we should go and talk about this face-to-face sometime before resolving this issue.

Reply:
The `Builder` should be `static` yes. 

I'm not sure I understand why creating a constructor for the Builder is non-trivial. What do you mean a `new ResourceDictionaries()` can be "unitialized"? If we don't want to build the `ResourceDictionaries` in the `Builder` constructor, we can pass it in:

```java
 LuthierIndustrialPark.Builder builder = new LuthierIndustrialPark.Builder(new LuthierResourceDictionaries());
```

or
```java
 LuthierResourceDictionaries resourceDictionaries = new LuthierResourceDictionaries();
  // prepolulate resourceDictionaries
 LuthierIndustrialPark.Builder builder = new LuthierIndustrialPark.Builder(resourceDictionaries);
```

I don't see what the problem is with any of this.

Now, maybe my typing was too general. The `LuthierIndustrialPark.Builder` should probably take and/or build a `LuthierResourceDictionaries` object. I'd forgotten that needed to exist when I was whipping up the `Builder` logic.

Reply:
The previous comment was made as I was not aware of a particular Java syntax. Now I have modified it so it no longer has such problem.